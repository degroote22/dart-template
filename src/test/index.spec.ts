import codegen from "../index";
import { readFileSync } from "fs";

describe("codegen", () => {
  it("works", () => {
    const args = JSON.parse(readFileSync("src/test/calls/1.json", "utf-8"));
    expect(
      codegen(args.templateContext, args.mergedDocuments, args.settings)
    ).toMatchSnapshot();
  });
});
