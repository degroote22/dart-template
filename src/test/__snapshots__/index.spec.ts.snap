// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`codegen works 1`] = `
Array [
  Object {
    "content": "



/// A repository contains the content for a project.
class ReadRepositoriesUserRepositoryConnectionRepository {
ReadRepositoriesUserRepositoryConnectionRepository(this._d);

final Map<String, dynamic> _d;

String get id => this._d[\\"id\\"];
/// The name of the repository.
String get name => this._d[\\"name\\"];
/// Returns a boolean indicating whether the viewing user has starred this starrable.
bool get viewerHasStarred => this._d[\\"viewerHasStarred\\"];

dynamic get rawData => this._d;

String toString() {
    return this._d.toString();
}
}

/// A list of repositories owned by the subject.
class ReadRepositoriesUserRepositoryConnection {
ReadRepositoriesUserRepositoryConnection(this._d);

final Map<String, dynamic> _d;

/// A repository contains the content for a project.
Option<List<Option<ReadRepositoriesUserRepositoryConnectionRepository>>> get nodes => option(this._d[\\"nodes\\"] != null, (this._d[\\"nodes\\"] as List).map<Option<ReadRepositoriesUserRepositoryConnectionRepository>>((repository) => option(repository!=null, ReadRepositoriesUserRepositoryConnectionRepository(repository))).toList());

dynamic get rawData => this._d;

String toString() {
    return this._d.toString();
}
}

/// A user is an individual's account on GitHub that owns repositories and can make new content.
class ReadRepositoriesUser {
ReadRepositoriesUser(this._d);

final Map<String, dynamic> _d;

/// A list of repositories owned by the subject.
ReadRepositoriesUserRepositoryConnection get repositories => ReadRepositoriesUserRepositoryConnection(this._d[\\"repositories\\"]);

dynamic get rawData => this._d;

String toString() {
    return this._d.toString();
}
}

class ReadRepositories {
ReadRepositories(this._d);

final Map<String, dynamic> _d;

/// A user is an individual's account on GitHub that owns repositories and can make new content.
ReadRepositoriesUser get viewer => ReadRepositoriesUser(this._d[\\"viewer\\"]);

dynamic get rawData => this._d;

String toString() {
    return this._d.toString();
}
}

",
    "filename": "ReadRepositoriesQuery.dart",
  },
]
`;
